projects/packages/classic-theme-helper/src/social-links.php

```diff
@@ -0,0 +1,279 @@
+ */
+
+namespace Automattic\Jetpack\Classic_Theme_Helper;
+
+// phpcs:disable Universal.Files.SeparateFunctionsFromOO.Mixed -- TODO: Move classes to appropriately-named class files.
+
+// @todo Un-comment and possibly move initialization when requiring the file.
+// if ( ! function_exists( 'jetpack_theme_supports_social_links' ) ) {
+// **
+// * Init Social_Links if the theme declares support.
+// *
+// * @suppress PhanNoopNew
+// */
+// function jetpack_theme_supports_social_links() {
+// if ( ! wp_is_block_theme() && current_theme_supports( 'social-links' ) && function_exists( 'publicize_init' ) ) {
+// new \Automattic\Jetpack\Classic_Theme_Helper\Social_Links();
+// }
+// }
+// add_action( 'init', 'jetpack_theme_supports_social_links', 30 );
+// }
+
+if ( ! class_exists( __NAMESPACE__ . '\Social_Links' ) ) {
+
+	/**
+	 * Social_Links main class.
+	 */
+	class Social_Links {
+
+		/**
+		 * The links the user set for each service.
+		 *
+		 * @var array
+		 */
+		private $links;
+
+		/**
+		 * A Publicize object.
+		 *
+		 * @var Publicize
+		 */
+		private $publicize;
+
+		/**
+		 * An array with all services that are supported by both Publicize and the
+		 * currently active theme.
+		 *
+		 * @var array
+		 */
+		private $services = array();
+
+		/**
+		 * An array of the services the theme supports
+		 *
+		 * @var array
+		 */
+		private $theme_supported_services = array();
+
+		/**
+		 * Constructor.
+		 */
+		public function __construct() {
+			$theme_support = get_theme_support( 'social-links' );
+
+			/*
+			An array of named arguments must be passed as the second parameter
+			 * of add_theme_support().
+			 */
+			if ( empty( $theme_support[0] ) ) {
+				return;
+			}
+
+			$this->theme_supported_services = $theme_support[0];
+			$this->links                    = class_exists( \Jetpack_Options::class ) ? \Jetpack_Options::get_option( 'social_links', array() ) : '';
+
+			$this->admin_setup();
+
+			add_filter( 'jetpack_has_social_links', array( $this, 'has_social_links' ) );
+			add_filter( 'jetpack_get_social_links', array( $this, 'get_social_links' ) );
+
+			foreach ( $theme_support[0] as $service ) {
+				add_filter( "pre_option_jetpack-$service", array( $this, 'get_social_link_filter' ) ); // - `get_option( 'jetpack-service' );`
+				add_filter( "theme_mod_jetpack-$service", array( $this, 'get_social_link_filter' ) ); // - `get_theme_mod( 'jetpack-service' );`
+			}
+		}
+
+		/**
+		 * Init the admin setup.
+		 */
+		public function admin_setup() {
+			if ( ! current_user_can( 'manage_options' ) ) {
+				return;
+			}
+
+			if ( ! is_admin() && ! $this->is_customize_preview() ) {
+				return;
+			}
+
+			// @phan-suppress-next-line PhanUndeclaredFunction -- Function checked with function_exists - see https://github.com/phan/phan/issues/1204.
+			$this->publicize    = function_exists( 'publicize_init' ) ? publicize_init() : null;
+			$publicize_services = $this->publicize->get_services( 'connected' );
```

Comment:
What happens here if `$this->publicize` is null?

Reply:
Good question. I've no idea at this stage, but I'll make sure to look into that when requiring the package in the next PR.